frame/contracts/Cargo.toml

```
@@ -77,3 +77,6 @@ try-runtime = ["frame-support/try-runtime"]
 # Make contract callable functions marked as __unstable__ available. Do not enable
 # on live chains as those are subject to change.
 unstable-interface = []
+
+[package.metadata.frame]
+pallet-version = 4
```

If we store this item in the pallet as `stoarge-version`, i think the name matching here would make sense.

```suggestion
storage-version = 4
```

---

Should this be `3` to match the existing pallet version?

---

I can do this, but then I would rename the struct etc to `StorageVersion` as well?

---

we use pallet version everywhere, in for `PalletVersion::storage_version` should be understood the pallet version in storage.

---

> Should this be `3` to match the existing pallet version?

I'm not sure, in the code if version is 3 we do an upgrade, I guess next release of contract should bump the version in order not to trigger the upgrade on every runtime upgrade. (though this runtime upgrade is harmless)

---

> we use pallet version everywhere, for me `PalletVersion::storage_version` should be understood the pallet version in storage.

So what you is your favorite naming?

---

I think the naming should make clear that it is about **storage** compatibility. `PalletVersion` simply makes me ask myself why it has to exist when we have the perfectly fine crate version. If you ask me I would completely remove everything that refers to it as a pallet version:

`StorageVersion::current_version(), StorageVersion::stored_version()`.

---

> So what you is your favorite naming?

considering we always refer to it as pallet version, and that I understand `PalletVersion::storage_version` as the pallet version in storage, then my favorite is to use `pallet-version` in the Cargo.toml here.

But if we rename everything to storage-version I don't mind either.